### ISSUE TITLE
Missing parts of documentation (Reverse Proxy)

### ISSUE THREAD

**Issue by nickheyer** [distribution/distribution#4475]
### Description

Registry is such a great resource for almost any scale of operations, but the documentation is seriously lacking - and in the below case, nonsensical.

For [example](https://distribution.github.io/distribution/recipes/nginx/#solution):

```
"We also implement push restriction (to a limited user group) for the sake of the example. Again, you should modify this to fit your mileage."
```

Push restrictions are not implemented for any users created in the example. When taken literally, this is potentially creating vulnerable container registries.

The provided solution for nginx auth needs to be reviewed, and likely expanded to include the mentioned solution. Extending upon this, it doesn't make sense to provide an example of a basic auth with `htpasswd` when the stated "use case" for nginx is to implement a SSO. Provide example nginx configurations for various SSO providers. 

### Reproduce

Go to: https://distribution.github.io/distribution/recipes/nginx/#solution

### registry version

lts

**Comment by milosgajdos** [comment]
> Push restrictions are not implemented for any users created in the example. When taken literally, this is potentially creating vulnerable container registries.

I think what the docs mean is that whoever wants to push in this set up needs to be able to pass the auth. I haven't written the docs so I'm hypothesizing here. We are open to PRs if you think you have a better way of phrasing this

**Comment by nickheyer** [comment]
> > Push restrictions are not implemented for any users created in the example. When taken literally, this is potentially creating vulnerable container registries.
> 
> I think what the docs mean is that whoever wants to push in this set up needs to be able to pass the auth. I haven't written the docs so I'm hypothesizing here. We are open to PRs if you think you have a better way of phrasing this

I don't believe that is the case. More likely a copy-paste mistake from the apache example just above it, where they did actually implement the scoped auth.
